### PR TITLE
feat: add option group to the currency dropdown

### PR DESCRIPTION
**Pull request by w3ea**
## Summary
https://app.clickup.com/t/p5cafr
https://app.clickup.com/t/pbh4c5

## Checklist

-   [x] My changes look good in both light AND dark mode
-   [x] The change is not hardcoded to a single network, but has multi-asset in mind
-   [ ] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: src/app/components/SelectDropdown/SelectDropdownItem.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		item,
+		index,
+		getItemProps,
+		highlightedIndex,
+		inputValue,
+		onMouseDown,
+		renderLabel,
+	}: SelectDropdownItemProperties) => (
+		<li
+			key={`${item.value}${index}`}
```
Comment (alesmit) [inline src/app/components/SelectDropdown/SelectDropdownItem.tsx:17]: We could add a separator between value and index to cover edge cases. For example `{ value: 1, index: 20 }` can produce the same key as `{ value: 12, index: 0 }`

File: src/app/components/SelectDropdown/styles.ts [hunk 2]
```diff
@@ -11,8 +11,23 @@ export const SelectOptionsList = styled.ul`
...
 	}
 
+	.select-list-option-group:not(:last-child) {
+		${tw`mb-3`};
+	}
+
+	.select-list-option-head,
+	.select-list-option {
+		${tw`relative px-10 border-0`};
+	}
```
Comment (alesmit) [inline src/app/components/SelectDropdown/styles.ts:21]: Is there any reason why we can't apply these tw classes directly on the element?

File: src/app/components/SelectDropdown/SelectDropdownItem.tsx (hunk 1 above)
Comment (w3ea) [reply src/app/components/SelectDropdown/SelectDropdownItem.tsx]: - we use text for value, and I think that rarely happens
- i guess these come from [downshift document](https://www.downshift-js.com/use-combobox#basic-usage)

p.s. i added a separator between value and index to make it more unique (👆🏻 these are just for your information)

File: src/app/components/SelectDropdown/styles.ts (hunk 2 above)
Comment (w3ea) [reply src/app/components/SelectDropdown/styles.ts:21]: we can do this, but for better readability and maintenance of code, we put everything in one file

**Review by faustbrian — APPROVED** [review]

**Pull request merged**